src/main/java/com/fpl/smdc/core/notification/config/AWSAppConfig.java

```diff
@@ -0,0 +1,144 @@
+    private String awsAccessKey;
+
+    @Value("${aws.secretKey}")
+    private String awsSecretKey;
+
+    @Value("${aws.region}")
+    private String awsRegion;
+
+    @SuppressWarnings("deprecation")
+	@Bean(name = "sessionCredentials")
+    public BasicAWSCredentials sessionCredentials() {
+
+        AWSSecurityTokenServiceClient sts_client = (AWSSecurityTokenServiceClient) AWSSecurityTokenServiceClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
+       // sts_client.withRegion(Regions.US_EAST_1);
+        GetSessionTokenRequest session_token_request = new GetSessionTokenRequest();
+      
+        if (this.credentialsValidityDuration == null || this.credentialsValidityDuration.trim().equals("")) {
+            session_token_request.setDurationSeconds(TEMPORARY_CREDENTIALS_DURATION_DEFAULT);
+        } else {
+            session_token_request.setDurationSeconds(Integer.parseInt(this.credentialsValidityDuration));
+        }
+
+        GetSessionTokenResult session_token_result = sts_client.getSessionToken(session_token_request);
+
+        Credentials session_creds = session_token_result.getCredentials();
+
+        BasicAWSCredentials  sessionCredentials = new BasicAWSCredentials (
+        		"AKIAZTLYPHZE2X72YYCK", "usTm5zefs3bQAnoqkGYGAoj9Ggsc/orPR28RAAVf");
+        return sessionCredentials;
+    }
+    @Autowired
+    private NotificationMessageHandlerMethodArgumentResolver notificationResolver;
+
+    @Override
+    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentResolvers) {
+        argumentResolvers.add(notificationMessageHandlerMethodArgumentResolver());
+    }
+
+    @Bean
+    public NotificationMessageHandlerMethodArgumentResolver notificationMessageHandlerMethodArgumentResolver () {
+
+        return new NotificationMessageHandlerMethodArgumentResolver();
+    };
+    
+    @Bean
+    public  NotificationMessagingTemplate notificationMessagingTemplate() {
+    	return new NotificationMessagingTemplate(amazonSNS());
+    }
+    
+    @Bean
+    public BonsaiSnsServer bonsaiSnsServer() {
+    	return new BonsaiSnsServer.Builder().withAccountId("660067008073").withRegion("us-east-1")
+        .withBonsaiSnsEnvironmentDefinition(
+            BonsaiSnsEnvironment.Companion.definition()
+                .withTopic(
+                    Topic.Companion.definition()
+                        .withName("LayerUpdate")
+                        .withSubscription(
+                            Subscription.Companion.definition()
+                                .withEndpoint("http://localhost:8080/sns/notification/subscirbe")
+                                .withProtocol("http")
+                    )
+                )
+        ).start();
+    }
+    
+    @Bean
+    public AmazonSNS amazonSNS () {
+    	
+
+          AWSCredentialsProvider awsCredentialsProvider = new AWSStaticCredentialsProvider(
+                  new BasicAWSCredentials(awsAccessKey, awsSecretKey)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).